microxrce_agent.cpp

```diff
@@ -103,25 +104,37 @@ int main(int argc, char** argv)
     {
         std::cout << "UDP agent initialization... ";
         uint16_t port = parsePort(cl[1]);
-#ifdef _WIN32
-        server = new eprosima::uxr::UDPServer(port);
-#else
-        server = (3 == cl.size()) //discovery port
+
+        server = 3 <= cl.size() && '-' != cl[2][0] //discovery port
```

Comment:
discovery_port shold only be read after `--discovery` flag found:

`agent 9999 --discovery` -> default discovery port used
`agent 9999 --discovery 8888` -> discovery on 8888
`agent 9999 8888` -> error
`agent 9999 8888` --discovery -> error

Reply:
Done